assets/js/common/ApiKeySettingsModal/ApiKeySettingsModal.jsx

```diff
@@ -0,0 +1,176 @@
+  const [timeQuantity, setTimeQuantity] = useState(0);
+  const [timeQuantityType, setTimeQuantityType] = useState(timeOptions[0]);
+
+  const [quantityError, setQuantityError] = useState(false);
+  const [timeFormEnabled, setTimeFormEnabled] = useState(true);
+
+  const saveFormState = () => {
+    if (!timeFormEnabled) {
+      onSave({ apiKeyExpiration: null });
+      return;
+    }
+    if (timeQuantity === 0 || !timeQuantity) {
+      setQuantityError(true);
+      return;
+    }
+    const timeQuantitySettings = availableTimeOptions.find(
+      (q) => q.type === timeQuantityType
+    );
+    const apiKeyExpiration = timeQuantitySettings.timeGenerator(timeQuantity);
+
+    onSave({ apiKeyExpiration: normalizeExpiration(apiKeyExpiration) });
+  };
+
+  return (
+    <Modal
+      title="API Key Settings"
+      className="!w-3/4 !max-w-3xl"
+      open={open}
+      onClose={onClose}
+    >
+      <div className="flex flex-col my-5">
+        <span className="my-2 mb-4 text-gray-500">
+          {' '}
+          By generating a new key, you will need to replace the API key on all
+          hosts.{' '}
+        </span>
+
+        <div className="flex space-x-1">
+          <div className="w-1/5">
+            <Label>Never Expires</Label>
+          </div>
+
+          <Switch
+            selected={!timeFormEnabled}
+            onChange={() => {
+              setTimeFormEnabled((enabled) => !enabled);
+              setQuantityError(false);
+            }}
+          />
+        </div>
+        <div className="flex items-center my-2 space-x-2">
+          <div className="w-1/3">
+            <Label>Key Expiration</Label>
+          </div>
+
+          <div className="w-2/4">
+            <Input
+              value={timeQuantity}
+              className="h-3/4"
+              type="number"
+              disabled={!timeFormEnabled}
+              error={quantityError}
+              onChange={({ target: { value } }) => {
+                setTimeQuantity(parseInt(value, 10));
+                setQuantityError(false);
+              }}
+            />
+          </div>
+          <div className="w-2/4 pt-4">
+            <Select
+              optionsName=""
+              options={timeOptions}
+              disabled={!timeFormEnabled}
+              value={timeQuantityType}
+              onChange={(value) => setTimeQuantityType(value)}
+            />
+          </div>
+          <div className="w-1/6 h-4/5">
+            <Button
+              onClick={() => saveFormState()}
+              disabled={quantityError || loading}
+            >
+              Generate
+            </Button>
+          </div>
+        </div>
+        {quantityError && (
+          <span className="my-1 mb-4 text-red-500">
+            {' '}
+            Key expiration value needs to be greater than 0{' '}
+          </span>
+        )}
+        {generatedApiKey && (
+          <div className="flex flex-col my-1 mb-4">
```

Comment:
Maybe this `div` could be subcomponent itself, as we use it (or should) the same content here in the modal and in the settings page.
What do you think?

Reply:
Anyway, if Jurgen agrees that they should be different, so be it

Reply:
If it helps us have consistency in the layout and implementation then I am happy for it to be a subcomponent.

Reply:
If it's a subcomponent, we will have the same copy button?